File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

//최대 10000*10000 이하면 가능
vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    for(int i=3; i<=2500; i++){ //가로
        for(int j=3; j<=2500; j++){ //세로
            if(((i-2)*(j-2) == yellow) && (brown+yellow) == i*j && i>=j){
                answer.push_back(i);
                answer.push_back(j);
                break;
            } 
        }
    }
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int total = brown + yellow;
        int h = 0;
        int w = 0;
        
        for(int i=1; i<15000; i++){
            if(total%i == 0){
                w = i;
                h = total/i;
                
                if((h*2+w*2-4) == brown){
                    answer[0] = h;
                    answer[1] = w;
                    break;
                }
            }
        }
        return answer;
    }
}